### PR TITLE
[JENKINS-41218] Use systemd for Linux installers

### PR DESCRIPTION
**Pull request by MarkEWaite**
## [JENKINS-41218](https://issues.jenkins.io/browse/JENKINS-41218) Use `systemd` for Linux installers instead of System V `init`

See [Jenkins developers mailing list discussion](https://groups.google.com/g/jenkinsci-dev/c/mRxFPGfuljA/m/y7Rb96V5AgAJ) for more details.

The Linux installers for Jenkins 2.335, 2.336, and 2.337 weekly releases use systemd instead of System V init.  They include migration steps that read the existing System V init script settings and convert them to systemd override configuration settings.

The systemd service has been the default and the preferred system and service manager on most Linux distributions for many years.  Jenkins use of systemd unifies the service management method across all three of our Linux installers and allows us to describe configuration actions for Debian, Red Hat, and SUSE installers with a single set of instructions.

Resolves the following known issues with the Linux installers:

* [JENKINS-41218](https://issues.jenkins.io/browse/JENKINS-41218) - Switch Linux installers from System V init to systemd
* [JENKINS-65809](https://issues.jenkins.io/browse/JENKINS-65809) - Restart the Jenkins service after plugin updates on Debian 11 (bullseye)
* [JENKINS-56219](https://issues.jenkins.io/browse/JENKINS-56219) - Restart systemd service when the controller exits unexpectedly
* [JENKINS-31656](https://issues.jenkins.io/browse/JENKINS-31656) - Return zero from RPM init script on successful stop
* [JENKINS-67487](https://issues.jenkins.io/browse/JENKINS-67487) - Correctly report startup result on Amazon Linux 2 installed with the rpm package
* [JENKINS-67404](https://issues.jenkins.io/browse/JENKINS-67404) - Do not fail startup with timeout on systems with slow DNS resolution

Changes include:

* Notify `systemd(1)` about start-up completion and other service status changes (#6228)
* Extend startup notification timeout as each initialization milestone is attained (#6237)
* Jenkins should terminate cleanly on `SIGTERM` (#6230)
* Note at-since from #6228 and #6237 (#6252)
* Update `@since` for 2.332.1

### Proposed changelog entries

* Use `systemd` based Linux installers instead of installers based on System V `init`.

### Proposed upgrade guidelines

Jenkins installers for Linux packages now use `systemd` to manage the configuration settings and the Jenkins service.  Settings from the previous System V based `init` installations are migrated to `systemd` during package upgrade.  Configuration setting changes are now performed with the command `systemctl edit jenkins` on all Linux packages.

### Submitter checklist

- [x] (If applicable) Jira issue is well described
- [x] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change) and are in the imperative mood. [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [x] Appropriate autotests or explanation to why this change has no tests
- [x] New public classes, fields, and methods are annotated with `@Restricted` or have `@since TODO` Javadoc, as appropriate. 
- [x] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@basil 

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [x] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [x] Changelog entries in the PR title and/or `Proposed changelog entries` are accurate, human-readable, and in the imperative mood
- [x] Proper changelog labels are set so that the changelog can be generated automatically
- [x] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [x] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins-ci.org/issues/?filter=12146)).

### CODE REVIEW

**Review by basil — APPROVED** [review]

**Review by timja — APPROVED** [review]

**Comment by basil** [comment]
This PR is now ready for merge. We will merge it after approximately 24 hours if there is no negative feedback. Please see the [merge process documentation](https://github.com/jenkinsci/jenkins/blob/master/docs/MAINTAINERS.adoc#merge-process) for more information about the merge process. Thanks!

**Pull request merged**
